clients/ts/signalr/src/LongPollingTransport.ts

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { AbortController } from "./AbortController";
+import { DataReceived, TransportClosed } from "./Common";
+import { HttpError, TimeoutError } from "./Errors";
+import { HttpClient, HttpRequest } from "./HttpClient";
+import { IConnection } from "./IConnection";
+import { ILogger, LogLevel } from "./ILogger";
+import { ITransport, TransferFormat } from "./ITransport";
+import { Arg, getDataDetail, sendMessage } from "./Utils";
+
+const SHUTDOWN_TIMEOUT = 5 * 1000;
+
+export class LongPollingTransport implements ITransport {
+    private readonly httpClient: HttpClient;
+    private readonly accessTokenFactory: () => string;
+    private readonly logger: ILogger;
+    private readonly logMessageContent: boolean;
+
+    private url: string;
+    private pollXhr: XMLHttpRequest;
+    private pollAbort: AbortController;
+    private shutdownTimeout: number;
+    private running: boolean;
+
+    constructor(httpClient: HttpClient, accessTokenFactory: () => string, logger: ILogger, logMessageContent: boolean) {
+        this.httpClient = httpClient;
+        this.accessTokenFactory = accessTokenFactory || (() => null);
+        this.logger = logger;
+        this.pollAbort = new AbortController();
+        this.logMessageContent = logMessageContent;
+    }
+
+    public connect(url: string, transferFormat: TransferFormat, connection: IConnection): Promise<void> {
+        Arg.isRequired(url, "url");
+        Arg.isRequired(transferFormat, "transferFormat");
+        Arg.isIn(transferFormat, TransferFormat, "transferFormat");
+        Arg.isRequired(connection, "connection");
+
+        this.url = url;
+
+        this.logger.log(LogLevel.Trace, "(LongPolling transport) Connecting");
+
+        // Set a flag indicating we have inherent keep-alive in this transport.
+        connection.features.inherentKeepAlive = true;
+
+        if (transferFormat === TransferFormat.Binary && (typeof new XMLHttpRequest().responseType !== "string")) {
+            // This will work if we fix: https://github.com/aspnet/SignalR/issues/742
+            throw new Error("Binary protocols over XmlHttpRequest not implementing advanced features are not supported.");
+        }
+
+        this.poll(this.url, transferFormat);
+        return Promise.resolve();
+    }
+
+    private async poll(url: string, transferFormat: TransferFormat): Promise<void> {
+        this.running = true;
+
+        const pollOptions: HttpRequest = {
+            abortSignal: this.pollAbort.signal,
+            headers: {},
+            timeout: 90000,
+        };
+
+        if (transferFormat === TransferFormat.Binary) {
+            pollOptions.responseType = "arraybuffer";
+        }
+
+        const token = this.accessTokenFactory();
```

Comment:
where's the await? Did you make these changes before all of the other changes?

Reply:
probably, I'll reconcile it

Reply:
Yes you did you basically don't have changes from the TS changes that were made. The accessToken factory and my change